### ISSUE TITLE
Can you describe the protocol used ?

### ISSUE THREAD

**Issue by X-Ryl669** [alexcustos/laano-tools#1]
Hi, 

 I've found your request for including LAANO in F-Droid and immediately found a similar software I have at home (using PHP). I don't use Nextcloud/Owncloud, so I'm more interested in knowning what is the protocol used for syncing the link/notes ? (I've read it's webdav, but this is too generic).

Can you describe the protocol a bit more precisely, so I can make a minimal PHP implementation of the server ?

Thanks.

**Comment by alexcustos** [comment]
Hello,

Yes you are right, the protocol is WebDAV. I chose Nextcloud library to communicate with shares, just because it was the best implementation at that moment and seems nothing changed for now. In all other respects, the protocol is about syncing simple JSON files with a remote file system. I thought it was a good idea because no special backend is needed and it's easy to make it work with other cloud platforms, but the obvious downside is performance.

I managed to make it work fast enough with some Nextcloud/ownCloud features over classical WebDAV, so it might not be working well with another WebDAV server. If you are going to use something else, it would be great if you confirm it's working at least somehow.

**Comment by X-Ryl669** [comment]
Thanks for answering.
What are the route used for storing, authenticating and fetching notes ?
I've found `/static/<path>` and `/save` in this repository, but I've no idea what to do with those.

In the main repository, I've found that it authenticates with the root URL (`ROOT_PATH = "/"`), but I don't see where you are fetching the notes from the server (only where it *uploads* them to the server).

Can you explain a bit more what the software is doing (with each request explained, and the expected results ?)

**Comment by alexcustos** [comment]
That's right. I'm sorry for the inconvenience. This app isn't meant to be a fully functional client. It because I didn't find the Python WebDAV library that allows working with file creation and modification timestamps. Without this option two-way syncing is practically impossible. I limited it for just uploading new records in order to push them to mobile clients.
